### PR TITLE
Webhook callback for stripe.disputes.created - which automatically closes them

### PR DESCRIPTION
**Pull request by cadecairos**
This patch adds a webhook callback, protected by a token, that stripe can hit when a dispute is created. The callback will close the dispute automatically.

@alicoding r?

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -71,6 +71,15 @@ module.exports = function(options) {
     uri: process.env.APPLICATION_URI
   });
 
+  server.register(require("hapi-auth-bearer-token"), function(err) {});
```
Comment (alicoding) [inline server.js]: we don't need to handle any error there? there could be an internal error there right?

File: server.js [hunk 2]
```diff
@@ -71,6 +71,15 @@ module.exports = function(options) {
...
+
+  server.auth.strategy("stripe", "bearer-access-token", {
+    validateFunc: function(token, callback) {
+      // this = request
```
Comment (alicoding) [inline server.js]: do we need `this`?

**Comment by alicoding** [comment]
travis is failing.

File: server.js (hunk 1 above)
Comment (cadecairos) [inline server.js]: incredibly, ridiculously unlikely, considering the module versions won't change unless manually changed

File: server.js (hunk 2 above)
Comment (cadecairos) [inline server.js]: nah, tis cruft

File: routes/index.js [hunk 3]
```diff
@@ -361,6 +361,24 @@ var routes = {
...
+
+    if (event.type !== 'charge.dispute.created') {
+      return reply('This hook only processes newly created disputes');
+    }
+
+    stripe.closeDispute(
+      event.data.object.id,
+      function(closeDisputeError, dispute) {
+        if (closeDisputeError) {
+          return reply(Boom.badImplementation('An error occurred while closing the dispute', closeDisputeError));
```
Comment (alicoding) [inline routes/index.js]: lowercase `boom`

**Comment by alicoding** [comment]
r+

**Pull request merged**